generator.go

```diff
@@ -123,24 +148,50 @@ func (g *Generator) Generate() (GenerateResult, error) {
 		g.Logger.Debugf("Finished running generate for extension %s", ext)
 	}
 
-	g.Logger.Debug("Validating Dockerfiles")
-	if err := g.validateDockerfiles(dockerfiles); err != nil {
+	g.Logger.Debug("Checking for new run image")
+	base, newBaseIdx, extend := g.checkNewRunImage(dockerfiles)
+	if err != nil {
 		return GenerateResult{}, err
 	}
+	if !containsMatch(g.RunMetadata.Images, base) {
+		return GenerateResult{}, fmt.Errorf("new runtime base image '%s' not found in run metadata", base)
+	}
 
 	g.Logger.Debug("Copying Dockerfiles")
-	if err := g.copyDockerfiles(dockerfiles); err != nil {
+	if err = g.copyDockerfiles(dockerfiles, newBaseIdx); err != nil {
 		return GenerateResult{}, err
 	}
 
-	g.Logger.Debug("Checking for new run image")
-	runImage, err := g.checkNewRunImage()
-	if err != nil {
-		return GenerateResult{}, err
+	newAnalyzedMD := g.AnalyzedMD
+	if newRunImage(base, g.AnalyzedMD) {
+		g.Logger.Debugf("Updating analyzed metadata with new run image '%s'", base)
+		newAnalyzedMD.RunImage = &platform.RunImage{ // target data is cleared
+			Reference: base,
+			Extend:    extend,
+		}
+	} else if extend && g.AnalyzedMD.RunImage != nil {
+		g.Logger.Debug("Updating analyzed metadata with run image extend")
+		newAnalyzedMD.RunImage.Extend = true
 	}
 
-	g.Logger.Debugf("Finished build, selected runImage '%s'", runImage)
-	return GenerateResult{Plan: filteredPlan, UsePlan: true, RunImage: runImage}, nil
+	return GenerateResult{
+		AnalyzedMD: newAnalyzedMD,
+		Plan:       filteredPlan,
+		UsePlan:    true,
+	}, nil
+}
+
+func containsMatch(images []platform.RunImageForExport, imageName string) bool {
```

Comment:
nit: given that len 0 is a wildcard match, is there a different word than "contains" ?  eg. `isMatch` ? `matches`? `satisfies`?

Reply:
Updated in b1c2c2e